[Odie] Call for testing fixes - 1

AllTerrainDeveloper opened the pull request:
Related to pdtkmj-20q-p2

## Proposed Changes

This is addressing a number of issues detected in the cFt. More Pull Request to come.

1. Full-screen mode for messages is now fully adjusted to the screen
2. Scrolling now is not being propagated to the parent while on full-screen
3. Inline help is now not changing its size while focused 
4. Removed semicolon after sources:` sources: => sources`
5. The Sources header is now clickable
6. The message is still rendered in the list of messages even after setting the full-screen mode for the message itself
7. Fixed a strange wobbling happening in Firefox while typing a long message.
8. Do not display again the contact options if we are already showing it as part of the message (downvoting an answer)
9. Fixed the display of long inline links.
10. Fixed copy/pasting the above texts that contained links, adding line breaks when they were not necessary
11. Disabled scroll to the bottom when typing in the message input box
12. For a split second an horizontal scroll appears in the full screen message while clicking on the sources
13. If message has been rated negatively, persist the contact options when reloading
14. Wapuu flows directly to email or chat when asked to do so (no more AI involved after Wapuu)

## Testing Instructions

Please, test as per the attached link

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

AllTerrainDeveloper committed:
Fixing Igor identified issues

AllTerrainDeveloper committed:
Igor - Full screen scroll and content behaviour

AllTerrainDeveloper committed:
Don't show other contact options if the message is already showing it
Fix scrolling wobbling when typing message

AllTerrainDeveloper committed:
Fix "Jump To Recent" button

AllTerrainDeveloper committed:
Fix: In Firefox for a split second a horizontal bar appears in the sources while in full screen

AllTerrainDeveloper committed:
Disliked feedback remains rendered with the contact options

AllTerrainDeveloper committed:
Persist negative feedback with its other contact options

AllTerrainDeveloper commented on client/odie/message/index.tsx:
```diff
@@ -79,6 +82,42 @@ const ChatMessage = ( { message, scrollToBottom }: ChatMessageProps ) => {
+	useEffect( () => {
+		const fullscreenElement = fullscreenRef.current;
+		if ( fullscreenElement ) {
+			fullscreenElement.addEventListener( 'wheel', handleWheel, { passive: false } );
+		}
+		return () => {
+			if ( fullscreenElement ) {
+				fullscreenElement.removeEventListener( 'wheel', handleWheel );
```
I needed to add the hook that way, as React does not offer a way of adding the "passive: false" optino

kthai15 committed:
source link updates

kthai15 committed:
remove unused component

kthai15 committed:
dedupe sources by url

kthai15 committed:
remove scroll on sources click

klimeryk commented on client/odie/message/style.scss:
```diff
@@ -123,9 +140,13 @@
 	margin: 0;
+
+	.odie-sources-link {
+		padding: 0 8px;
```
```suggestion
		padding: 0 4px;
```

IMHO, the 8px side padding is too much:
<img width="395" alt="Screenshot 2023-11-10 at 21 19 05" src="https://github.com/Automattic/wp-calypso/assets/3392497/145f874f-0e96-4296-a72b-fb1998deeef4">

4px looks more balanced on the sides:
<img width="399" alt="Screenshot 2023-11-10 at 21 20 18" src="https://github.com/Automattic/wp-calypso/assets/3392497/5450c281-9258-4871-a292-f2869145da8a">

kthai15 replied:
Thanks! Just updated

kthai15 committed:
update behavior of jump to recent button

this will now only show if the most recent message isn't in view, and will scroll to the start of the message instead of the end

klimeryk commented:
> Fixed a strange wobbling happening in Firefox while typing a long message.

It's indeed much improved, but there's a bit of a jump when a character is deleted:


https://github.com/Automattic/wp-calypso/assets/3392497/b5372b99-52b1-41f8-87d9-9b737a048334

kthai15 committed:
reduce padding for inline links

kthai15 committed:
check for empty sources

kthai15 committed:
make popout message header sticky

kthai15 commented:
> It's indeed much improved, but there's a bit of a jump when a character is deleted:

Thanks, I'll create an issue for this.

AllTerrainDeveloper committed:
Aditional fixes
Fix Wapuu flow
Fix message auto scrolling to the top of message with reaching bottom
Fix message lossing reference when exiting full screen

kthai15 reviewed (approved):
Tested all the issues minus the UX for HE escalation, but we'll let it up to testers and HappOps to determine what else needs to be done in that regard.

A LOT of improvements here, so we should shipit and have testers verify the fixes.
